plugins/outputs/arc/arc_test.go

```diff
@@ -0,0 +1,467 @@
+		// Read body
+		var body io.Reader = r.Body
+		if r.Header.Get("Content-Encoding") == "gzip" {
+			gz, err := gzip.NewReader(r.Body)
+			assert.NoError(t, err)
+			defer gz.Close()
+			body = gz
+		}
+
+		data, err := io.ReadAll(body)
+		assert.NoError(t, err)
+
+		// Try to decode as array of columnar data (multiple measurements)
+		reader := msgp.NewReader(bytes.NewReader(data))
+		decoded, err := reader.ReadIntf()
+		assert.NoError(t, err)
+
+		columnarDataArray := decoded.([]interface{})
+		assert.Len(t, columnarDataArray, 2, "should have 2 measurements")
+
+		// Verify we have both cpu and mem measurements
+		measurementNames := make(map[string]bool)
+		for _, item := range columnarDataArray {
+			colData := item.(map[string]interface{})
+			measurementNames[colData["m"].(string)] = true
+			assert.NotEmpty(t, colData["columns"])
+		}
+
+		assert.True(t, measurementNames["cpu"], "should have cpu measurement")
+		assert.True(t, measurementNames["mem"], "should have mem measurement")
+
+		w.WriteHeader(http.StatusNoContent)
+	}))
+	defer ts.Close()
+
+	a := &Arc{
+		URL:              ts.URL,
+		HTTPClientConfig: httpconfig.HTTPClientConfig{Timeout: config.Duration(5 * time.Second)},
+		ContentEncoding:  "identity",
+		Log:              testutil.Logger{},
+	}
+
+	err := a.Init()
+	require.NoError(t, err)
+
+	err = a.Connect()
+	require.NoError(t, err)
+
+	// Write metrics from different measurements
+	metrics := []telegraf.Metric{
+		testutil.MustMetric(
+			"cpu",
+			map[string]string{"host": "server01"},
+			map[string]interface{}{"usage_idle": float64(95.5)},
+			time.Unix(1633024800, 0),
+		),
+		testutil.MustMetric(
+			"mem",
+			map[string]string{"host": "server01"},
+			map[string]interface{}{"usage_percent": float64(75.0)},
+			time.Unix(1633024800, 0),
+		),
+	}
+
+	err = a.Write(metrics)
+	require.NoError(t, err)
+}
+
+func TestSampleConfig(t *testing.T) {
+	a := &Arc{}
+	cfg := a.SampleConfig()
+	require.NotEmpty(t, cfg)
+	require.Contains(t, cfg, "url")
+	require.Contains(t, cfg, "api_key")
+}
+
+func TestInit(t *testing.T) {
+	tests := []struct {
+		name        string
+		arc         *Arc
+		expectError bool
+	}{
+		{
+			name: "valid config",
+			arc: &Arc{
+				URL:              "http://localhost:8000/api/v1/write/msgpack",
+				HTTPClientConfig: httpconfig.HTTPClientConfig{Timeout: config.Duration(5 * time.Second)},
+			},
+			expectError: false,
+		},
+		{
+			name: "default values",
+			arc: &Arc{
+				URL: "",
+			},
+			expectError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.arc.Init()
+			if tt.expectError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestClose(t *testing.T) {
+	a := &Arc{
+		Log: testutil.Logger{},
+	}
+	err := a.Init()
+	require.NoError(t, err)
+
+	err = a.Connect()
+	require.NoError(t, err)
+
+	err = a.Close()
+	require.NoError(t, err)
+}
```

Comment:
Hmmm what does this test? I do have a hard time seeing where it could fail or what the benefit of having it is.